### PR TITLE
feat(webex-core): Refactor updateCredentialsConfig

### PR DESCRIPTION
**Pull request by AMANDUA**
# Pull Request Template
Refactor `updateCredentialsConfig`, to remove trailing slash if its there.
Recently found an issue, yesterday U2C is sending identity url as `https://identitybts.webex.com` and the following utility updates the url as below.
image

This was causing an actual issue in integration server for `onboarding`, though U2C has corrcted it today (Andrew updated me about that).

But for a strong check we must check for a `Slash` first and then remove it. Else this issue may happen again.

## Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

Fixes # (issue)

## Type of Change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Test Coverage

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Test A
- [ ] Test B

**Test Configuration**:
* SDK Version
* Node/Browser Version
* NPM Version

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by adamweeks** [comment]
Thanks for adding this feature. Could you add some unit tests that verify the broken case no longer exists? 

What we usually do is create a unit test that fails before the code change, then passes after the code change.

**Comment by arun3528** [comment]
looks good after the unit test

**Comment by AMANDUA** [comment]
Agreed @adamweeks , @arun3528 @chrisadubois .
I will add unit test and update you guys.

Thank You

**Comment by AMANDUA** [comment]
@adamweeks -  updated the Unit tests, please review

9b367b56-2ea1-4f64-a129-d5a609c4d205

added both cases one with `/` and one without `/`, and validated that only remove trailing `/` if exists

File: packages/node_modules/@webex/webex-core/test/unit/spec/services/services.js [hunk 1]
```diff
@@ -210,6 +213,32 @@ describe('webex-core', () => {
...
+        const servicesList = {
+          idbroker: 'https://idbroker.webex.com',
+          identity: 'https://identity.webex.com/'
+        };
+
+        catalog.list = sinon.stub().returns(servicesList);
+        await services.updateCredentialsConfig();
+      });
+
+      it('must update IDBROKER_BASE_URL as serviceList.idbroker', () => {
```
Comment (adamweeks) [inline packages/node_modules/@webex/webex-core/test/unit/spec/services/services.js:234]: To me, this reads as it is updating the environment variable. Perhaps a better naming structure for this test would be:
```suggestion
      it('sets the idbroker url properly when no trailing slash is present', () => {
```

File: packages/node_modules/@webex/webex-core/test/unit/spec/services/services.js [hunk 2]
```diff
@@ -210,6 +213,32 @@ describe('webex-core', () => {
...
+        assert.equal(webex.config.credentials.idbroker.url, expectedServiceList.idbroker);
+      });
+
+      it('must update IDENTITY_BASE_URL as serviceList.identity and remove trailing /', () => {
```
Comment (adamweeks) [inline packages/node_modules/@webex/webex-core/test/unit/spec/services/services.js:238]:
```suggestion
      it('sets the identity url properly when a trailing slash is present', () => {
```

**Comment by lalli-flores** [comment]
Looks good, thanks @AMANDUA

**Pull request merged**